### PR TITLE
feat(discord-bot): add "Feedback Command"

### PR DESCRIPTION
**Pull request by dhruv-kaushik**
# Description
Added the feedback Command but I need to know what needs to be done with the feedback..

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
N/A

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My PR title follows the [semantic commits](https://www.conventionalcommits.org/en/v1.0.0/) style

### CODE REVIEW

**Comment by RhysSullivan** [comment]
Thanks for opening a PR!

For tracking the feedback, you can use the trackDiscordEvent function

You'll need to update the analytics package to include the event

Have the event name be "Feedback" with the properties:

"source": "slash-command"
"canFollowUp": Boolean, set based on slash command input
"type": "bug" | "suggestion"

Here's an example https://github.com/AnswerOverflow/AnswerOverflow/blob/main/apps/discord-bot/src/listeners/events/on-question-ask.ts

I'll take a deeper look tomorrow, also if adding the analytics event is a pain don't worry about it I can jump in this PR and update it

**Comment by dhruv-kaushik** [comment]
I updated the types in events.ts what Wording i came up with is
advisor is the person who actually gave the feedback I couldn't think of a better word for that
```ts
type FeedbackGivenProps = UserProps<'User'> & {
	FeedbackType: 'Bug' | 'Suggestion';
	Advisor: UserProps<'advisor'>;
	CanFollowUp: boolean;
};
```

File: apps/discord-bot/src/utils/analytics.ts [hunk 1]
```diff
@@ -114,6 +115,12 @@ export type QuestionSolvedProps = QuestionAskedProps &
 type MarkSolutionUsedProps = UserProps<'User'> & {
 	Status: MarkSolutionErrorReason | 'Success';
 };
+type FeedbackGivenProps = ServerPropsWithDiscordData &
+		ChannelPropsWithDiscordData & 
+		UserProps<'advisor'> & {
+	FeedbackType: 'Bug' | 'Suggestion';
+	CanFollowUp: boolean;
+};
```
Comment (dhruv-kaushik) [inline apps/discord-bot/src/utils/analytics.ts:123]: I want to know why we need source property in this ?

Comment (RhysSullivan) [reply apps/discord-bot/src/utils/analytics.ts:123]: > why we need source property 

In the future, we'll be using this feedback event for multiple locations so the source will distinguish between them

>  also are serverProps and ChannelProps needed in this type

Sure lets include them, it'll be helpful to know where the feedback is coming from

> feedbackGiver

Sure, that works for it

**Comment by dhruv-kaushik** [comment]
@RhysSullivan, i can't figure out stuff in events so just take over

**Comment by RhysSullivan** [comment]
@dhruv-kaushik will do, I'll get this merged tomorrow

**Pull request closed without merging**